### PR TITLE
Add binary-search exercise (#297)

### PR DESCRIPTION
**Pull request by Adrien-LUDWIG**
Related to #297

I put 2 as difficulty, what do you think?

Is the prototype and naming good (i.e. `binary-search array value`), or should it follow the instruction story (i.e. `find-song-index playlist song`)?

Do not hesitate to tell me there are things to modify @BNAndras. 😉

### CODE REVIEW

File: exercises/practice/binary-search/.meta/config.json
Comment (BNAndras) [inline exercises/practice/binary-search/.meta/config.json:1]: Looks good, but could you run `./bin/configlet fmt -uye binary-search`? That'll format the json nicely.

File: exercises/practice/binary-search/binary-search-test.rkt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+     (test-exn "a value larger than the array's smallest value is not found"
+               exn:fail?
+               (lambda () (binary-search (list 1 3 4 6 8 9 11) 13)))
+     (test-exn "nothing is found in an empty array"
+               exn:fail?
+               (lambda () (binary-search '() 1)))
+     (test-exn "nothing is found when the left and right bounds cross"
+               exn:fail?
+               (lambda () (binary-search (list 1 2) 0)))
+     ))
```
Comment (BNAndras) [inline exercises/practice/binary-search/binary-search-test.rkt:45]: Can you move these two parentheses on the previous line? It's a general Lisp convention to "collect" all your trailing parentheses on the last line. Parentheses are there for the compiler, not the reader, who likely would benefit more from the indentations.

**Review by BNAndras — CHANGES_REQUESTED** [review]

**Comment by BNAndras** [comment]
> I put 2 as difficulty, what do you think?
Seems fine. The website bins difficulties 1 - 3 as "easy" so the specific level isn't too important. I try to avoid level 1 simply because exercises at the same level should be roughly equivalent in difficulty. `hello-world` is level 1 so only pretty straightforward exercises should also be level 1 in my opinion.

> Is the prototype and naming good (i.e. binary-search array value), or should it follow the instruction story (i.e. find-song-index playlist song)?
The naming is fine. Typically, I'd go with the property name from the canonical test json if it makes sense (so `find array value`). If the name can be tweaked to be more idiomatic, I'd do that like when transforming `a` to `b`, you'll often see `a->b` like `number->string'. However `binary-search` is fine, and other tracks take that approach as well.

**Comment by Adrien-LUDWIG** [comment]
Ok, thanks!

Well, I think it's ready then. I'll wait for your review and try to work on another in the meantime. 😉

**Comment by Adrien-LUDWIG** [comment]
Sorry, I didn't see your review on my phone. The formatting issues are resolved.

Do not hesitate to tell me if there's anything else.

File: exercises/practice/binary-search/binary-search-test.rkt
Comment (BNAndras) [inline exercises/practice/binary-search/binary-search-test.rkt:1]: For defining the lists for the tests, I'd prefer [using the quote form](https://docs.racket-lang.org/guide/quote.html) so `\'(...)` not `(list ...)`. That avoids evaluating the expressions inside the form so it's pretty common for lists. You could just do a find + replace within this file. That's it for the review. Thanks again for porting this exercise.

**Comment by Adrien-LUDWIG** [comment]
Done. 😉

**Comment by Adrien-LUDWIG** [comment]
Oh, and could you please add the `hacktoberfest-accepted` label? 🙏😄

**Review by BNAndras — APPROVED** [review]

**Pull request merged**